### PR TITLE
Lowered 'Found overlapping blocks during compaction' log level

### PR DESCRIPTION
**Pull request by pracucci**
In Cortex (and Thanos) having overlapping blocks is expected. Our users get frequently confused by this warning log message, which make them think it's an unexpected condition.

What's the sentiment to lower the log level? Or should I introduce an option to `NewLeveledCompactor()` to specify whether overlapping blocks are expected?

### CODE REVIEW

**Comment by roidelapluie** [comment]
It makes a lot of sense in Prometheus too since we have backfill now.

**Review by roidelapluie — APPROVED** [review]

**Review by beorn7 — APPROVED** [review]
LGTM

@codesome @bwplotka @brian-brazil would be great to get a maintainer's approval. I'd like to have this in the v2.24 release happening in two days.

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**
